src/router/websocket/router.ts

```diff
@@ -0,0 +1,191 @@
+import { trimSlash } from '../utils.ts';
+import { Constructor } from '../../utils/constructor.ts';
+import { hookName } from '../../hook/interfaces.ts';
+import {
+	FilterExecutor,
+	GuardExecutor,
+	HttpContext,
+	Injector,
+} from '../../mod.ts';
+import {
+	Application,
+	getPath,
+	HonoRequest,
+	RegExpRouter,
+	SmartRouter,
+	TrieRouter,
+} from '../../deps.ts';
+import { MetadataHelper } from '../../metadata/helper.ts';
+import {
+	ControllerConstructor,
+	ExecutionContext,
+	MiddlewareExecutor,
+} from '../mod.ts';
+import { resolveMethodParam } from '../controller/params/resolver.ts';
+
+export class WebSocketRouter {
+	private middlewareExecutor: MiddlewareExecutor;
+	constructor(
+		private injector: Injector,
+		private guardExecutor: GuardExecutor = new GuardExecutor(injector),
+		private filterExecutor: FilterExecutor = new FilterExecutor(injector),
+		private router: Application,
+	) {
+		this.middlewareExecutor = new MiddlewareExecutor(
```

Comment:
For consistency, I think we can move this to the `cconstructor` params, like `guardExecutor` and `filterExecutor`